File: fib.rb
#!/usr/bin/env ruby

def fib(n)
  (n < 2 && 1) || fib(n - 1) + fib(n - 2)
end

if $0 == __FILE__
  n = (ARGV[0] || 40).to_i
  puts fib(n)
end

File: fib.js
#!/usr/bin/env node

'use strict';

function fib(n) {
  if (n < 2) {
    return 1;
  } else {
    return fib(n - 1) + fib(n - 2);
  }
}

var n = 40;
if (process.argv.length > 2) {
  n = parseInt(process.argv[2]);
}

console.log(fib(n));
